src/neptune/metadata_containers/metadata_container.py

```diff
@@ -143,6 +148,37 @@ def __init__(
 
         self._startup(debug_mode=mode == Mode.DEBUG)
 
+        os.register_at_fork(after_in_child=self._handle_fork_in_child, after_in_parent=self._handle_fork_in_parent)
```

Comment:
You can use `reset_internal_ssl_state` directly here (IDE suggestion)

Reply:
I don't like this suggestion. :)